src/astx/flows.py

```diff
@@ -418,3 +418,104 @@ def get_struct(self, simplified: bool = False) -> ReprStruct:
         }
 
         return self._prepare_struct(key, value, simplified)
+
+
+@public
+@typechecked
+class CaseStmt(StatementType):
+    """AST class for a branch in a switch/match expression."""
+
+    condition: Optional[Expr] = None
+    body: Expr
+    default: bool = False
+
+    def __init__(
+        self,
+        body: Expr,
+        condition: Optional[Expr] = None,
+        default: bool = False,
+        loc: SourceLocation = NO_SOURCE_LOCATION,
+        parent: Optional[ASTNodes] = None,
+    ) -> None:
+        """Initialize the CaseStmt instance."""
+        super().__init__(loc=loc, parent=parent)
+        self.condition = condition
+        self.body = body
+        self.default = default
+        self.kind = ASTKind.CaseStmtKind
+
+        if self.default is False and self.condition is None:
+            raise ValueError(
+                "Condition must be provided for non-default branches."
+            )
+
+        if self.default is True and self.condition is not None:
+            raise ValueError(
+                "Condition must NOT be provided for default branches."
+            )
+
+    def __str__(self) -> str:
+        """Return a string representation of the object."""
+        return f"CaseStmt[{self.condition} => {self.body}]"
+
+    def get_struct(self, simplified: bool = False) -> ReprStruct:
+        """Return the AST structure of the object."""
+        key = "CASE-STMT"
+        condition_dict = (
```

Comment:
we can discuss more about this ..
i will share another idea for the record, so we can discuss about that later

```python
        branch = (
            {"default": "True"}
            if self.condition is None
            else {"condition": self.condition.get_struct(simplified)}
        )
```

not sure if it would actually work

Reply:
Yes, we can discuss this. I'll just put a couple of images here for reference:
![switch_get_struct_1](https://github.com/user-attachments/assets/6d09ed48-0bd4-47e4-9f81-cb905c6d6517)
![switch_get_struct_2](https://github.com/user-attachments/assets/3168eaa5-0cae-4904-9f11-4b2026c954bf)